common/djangoapps/course_modes/models.py

```diff
@@ -367,6 +367,91 @@ def min_course_price_for_currency(cls, course_id, currency):
         modes = cls.modes_for_course(course_id)
         return min(mode.min_price for mode in modes if mode.currency == currency)
 
+    @classmethod
+    def get_certificate_display(cls, mode, verification_status):
+        """
+         on the basis of mode and verification status make certificate display html strings and css class.
+         Args:
+            mode (str): enrollment mode.
+            verification_status (str) : verification status of student
+
+        Returns:
+            dictionary:
+        """
+
+        # import inside the function to avoid the circular import
+        from student.helpers import (
+            VERIFY_STATUS_NEED_TO_VERIFY,
+            VERIFY_STATUS_SUBMITTED,
+            VERIFY_STATUS_APPROVED
+        )
+
+        show_image = False
+        enrollment_title, enrollment_value, image_alt = '', '', ''
+
+        if mode == "verified":
+            if verification_status in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED]:
+                enrollment_title = "Your verification is pending"
+                enrollment_value = "Verified: Pending Verification"
+                show_image = True
+                image_alt = "ID verification pending"
+            elif verification_status == VERIFY_STATUS_APPROVED:
+                enrollment_title = "You're enrolled as a verified student"
+                enrollment_value = "Verified"
+                show_image = True
+                image_alt = "ID Verified Ribbon/Badge"
+            else:
+                enrollment_title = "You're enrolled as an honor code student"
+                enrollment_value = "Honor Code"
+        elif mode == "honor":
+            enrollment_title = "You're enrolled as an honor code student"
+            enrollment_value = "Honor Code"
+        elif mode == "audit":
+            enrollment_title = "You're auditing this course"
+            enrollment_value = "Auditing"
+        elif mode == "professional" or mode == "no-id-professional":
+            enrollment_title = "You're enrolled as a professional education student"
+            enrollment_value = "Professional Ed"
+
+        return {
+            'enrollment_title': enrollment_title,
+            'enrollment_value': enrollment_value,
+            'show_image': show_image,
+            'image_alt': image_alt,
+            'mode_class': CourseMode._get_certificate_css_class(mode, verification_status)
+        }
```

Comment:
This is the right idea, but I'm concerned about including the CSS class in this dictionary.  This can be a problem for maintainability, since the UX team will usually look in the templates to determine which CSS classes are used.  It would be more maintainable to pass back an enumerated value describing which ribbon to display, then let the templates decide which CSS class to use.